src/overlays/actors/ovl_En_Lift_Nuts/z_en_lift_nuts.c

```diff
@@ -28,111 +61,921 @@ ActorInit En_Lift_Nuts_InitVars = {
+    if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+        if ((this->textId == 0x27EE) || (this->textId == 0x27F4) || (this->textId == 0x27F5)) {
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 10);
+        } else {
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 0);
+        }
+        this->actionFunc = func_80AEACF8;
+    }
+}
+
+void func_80AEACF8(EnLiftNuts* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    switch (Message_GetState(&play->msgCtx)) {
+        case 0:
+        case 1:
+        case 2:
+        case 3:
+            break;
+        case 4:
+            func_80AEA7A4(this, play);
+            break;
+        case 5:
+            func_80AEA910(this, play);
+            break;
+        case 6:
+            if (Message_ShouldAdvance(play)) {
+                player->stateFlags1 &= ~PLAYER_STATE1_20;
+                func_80AE9FC8(this);
+                func_80AE9AC4(this, 2);
+                if (func_80AE9B4C(0, 3)) {
+                    func_80AE9B4C(1, 0);
+                }
+            }
+            break;
+    }
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0xA, 0x2000, 0x500);
+    if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+        switch (this->textId) {
+            case 0x27EE:
+                Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 11);
+                break;
+            case 0x27EF:
+                Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 1);
+                break;
+            case 0x27F4:
+                if (this->unk_354 == 0) {
+                    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 11);
+                    this->unk_354++;
+                } else if (this->unk_354 == 4) {
+                    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 12);
+                    this->unk_354 = 0;
+                } else {
+                    this->unk_354++;
+                }
+                break;
+        }
+    }
+    func_80AEB974(this);
+}
+
+void func_80AEAEAC(EnLiftNuts* this) {
+    this->actor.speedXZ = 2.0f;
+    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 1);
+    func_80AE9AC4(this, 1);
+    func_80AE9B4C(1, 1);
+    this->actionFunc = func_80AEAF14;
+}
+
+void func_80AEAF14(EnLiftNuts* this, PlayState* play) {
+    f32 dist;
+
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.home.rot.y, 10, 0x1000, 0x500);
+    dist = Math_Vec3f_StepTo(&this->actor.world.pos, &this->vec_1D8, this->actor.speedXZ);
+    this->actor.world.pos.y += this->actor.gravity;
+
+    if (dist == 0.0f) {
+        func_80AEAF8C(this);
+    }
+}
+
+void func_80AEAF8C(EnLiftNuts* this) {
+    this->actionFunc = func_80AEAFA0;
+}
+
+void func_80AEAFA0(EnLiftNuts* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    f32 dist;
+    f32 fv0;
```

Comment:
```suggestion
    f32 temp_fv0;
```
If we're not going to name it better to make it obvious.